hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirectory.java

```diff
@@ -1468,6 +1475,30 @@ static Collection<String> normalizePaths(Collection<String> paths,
     return normalized;
   }
 
+  /**
+   * Get the first Node that sets Quota.
+   */
+  static INode getFirstSetQuotaParentNode(INodesInPath iip) {
+    for (int i = iip.length() - 1; i > 0; i--) {
+      INode currNode = iip.getINode(i);
+      if (currNode == null) {
```

Comment:
Will it meet null here? if it is not expected, we should throw exception IMO.

Reply:
Here we traverse from the last INode to the penultimate node, excluding the root node.

Reply:
There should be no expected

Reply:
If it is not expected, we need to throw exception, or I think no need to check it.

Reply:
If that is true, we should loop begin from `iip.length() - 2`?